src/app/dashboard/component/organism-details/organism-details.component.ts

```diff
@@ -408,70 +417,353 @@ export class OrganismDetailsComponent implements OnInit, AfterViewInit {
   }
 
   getFiltersForSelectedFilter(data: any) {
-    const filters = {
+    const metadataFilters = {
+      sex: {},
+      trackingSystem: {},
+      organismPart: {}
+    };
+    const symbiontsFilters = {
+      sex: {},
+      trackingSystem: {},
+      organismPart: {}
+    };
+    const metagenomeFilters = {
+      sex: {},
+      trackingSystem: {},
+      organismPart: {}
+    };
+    this.metadataSexFilters = [];
+    this.metadataTrackingSystemFilters = [];
+    this.metadataOrganismPartFilters = [];
+
+    this.symbiontsSexFilters = [];
+    this.symbiontsTrackingSystemFilters = [];
+    this.symbiontsOrganismPartFilters = [];
+
+    this.metagenomesSexFilters = [];
+    this.metagenomesTrackingSystemFilters = [];
+    this.metagenomesOrganismPartFilters = [];
+
+    this.metadataFilters = metadataFilters;
+    this.metagenomeFilters = metagenomeFilters;
+    this.metagenomeFilters = metagenomeFilters;
+    for (const item of data) {
+      if (item.sex != null) {
+        if (item.sex in metadataFilters.sex) {
+          metadataFilters.sex[item.sex] += 1;
+        } else {
+          metadataFilters.sex[item.sex] = 1;
+        }
+      }
+      if (item.trackingSystem != null) {
+        if (item.trackingSystem in metadataFilters.trackingSystem) {
+          metadataFilters.trackingSystem[item.trackingSystem] += 1;
+        } else {
+          metadataFilters.trackingSystem[item.trackingSystem] = 1;
+        }
+      }
+
+      if (item.organismPart != null) {
+        if (item.organismPart in metadataFilters.organismPart) {
+          metadataFilters.organismPart[item.organismPart] += 1;
+        } else {
+          metadataFilters.organismPart[item.organismPart] = 1;
+        }
+      }
+
+      this.metadataFilters = metadataFilters;
+      const sexFilterObj = Object.entries(this.metadataFilters.sex);
+      const trackFilterObj = Object.entries(this.metadataFilters.trackingSystem);
+      const orgFilterObj = Object.entries(this.metadataFilters.organismPart);
+      const j = 0;
+      // tslint:disable-next-line:prefer-for-of
+      for (let i = 0; i < sexFilterObj.length; i++) {
```

Comment:
Instead of using for loop with an index variable and disabling tslint, replace it with for of, e.g 
for (const filter of sexFilterObj) {
  const jsonObj = { key: filter[j], doc_count: filter[j + 1] };
  this.metadataSexFilters.push(jsonObj);
}